### PR TITLE
doc/ceph.io: discover -> technology (1 of x)

### PR DESCRIPTION
**Pull request by zdover23**
This is the first in a series of PRs that will
rewrite the "Discover -> Technology" page. This
page adds meaningful text and links to documentation
that more completely explains the components referred
to on this page.

Lorem ipsum text remains present here. When the technical
content added in this PR is approved, I'll ask Pete to remove
the lorem ipsum text.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

### CODE REVIEW

File: src/en/discover/technology/index.html [hunk 1]
```diff
@@ -9,14 +9,9 @@ <h1 class="h1 visually-hidden">{{ title }}</h1>
 <section class="bg-navy section section--full">
   <div class="wrapper">
     <div class="max-w-224 mx-auto py-16 text-center">
-      <h2 class="color-white h1">Infinitely scalable data storage for object, block and file&nbsp;types.</h2>
+      <h2 class="color-white h1">Ceph uniquely delivers object, block, and file storage in a single, unified system.</h2>
```
Comment (liewegas) [inline src/en/discover/technology/index.html:12]: drop 'uniquely'

Comment (zdover23) [reply src/en/discover/technology/index.html]: "uniquely" has been dropped.

File: src/en/discover/technology/index.html [hunk 2]
```diff
@@ -58,15 +53,19 @@ <h2 class="h2 mb-12 lg:mb-16" id="object">Object storage</h2>
...
     <div>
       <p class="standout">
-        For businesses requiring seamless access to objects, Ceph provide comprehensive storage compatible with popular interfaces including
-        Amazon S3.
+      The Ceph Object Storage daemon (radosgw) is a FastCGI service that
+      provides a RESTful HTTP API to store objects and metadata. You can use
+      either the OpenStack Swift-compatible API or the Amazon S3-compatible
+      API, because the RADOS Gateway muses a unified namespace. This means that
+      you can write S3-compatible data with one application and read the same
+      data with another application that uses the Swift-compatible API.
```
Comment (liewegas) [inline src/en/discover/technology/index.html:61]: The Ceph RGW object storage service provides industry-leading S3 API compatibility with a robust set of security, tiering, and interoperability features.  Applications written to use S3 or Swift object storage can take advantage of Ceph's scalability and performance within a single datacenter, or federate multiple Ceph clusters across the globe to create a global storage namespace with an extensive set of replication, migration, and other data services.

File: src/en/discover/technology/index.html [hunk 3]
```diff
@@ -114,21 +113,20 @@ <h2 class="h2 mb-12 lg:mb-16" id="block">Block storage</h2>
   <div class="grid lg:grid--cols-2 grid--gap-14 lg:grid--gap-28">
     <div>
       <p class="standout">
-        Access to block device images through Ceph’s thinly provisioned RADOS Block Device (RBD), with capabilities including snapshotting,
-        replication and striping for maximised availability.
+        Ceph Block Devices stripe block-device images over multiple objects in the Ceph Storage Cluster, where all objects are mapped to placement groups and distributed. The placement groups are then spread across separate ceph-osd daemons throughout the cluster. Thin-provisioned and snapshottable, Ceph Block Devices are an attractive option for cloud computing and virtualization.
```
Comment (liewegas) [inline src/en/discover/technology/index.html:116]: Ceph RBD (RADOS Block Device) block storage stripes virtual disks over objects in the Ceph storage cluster, distributing data and workload across all available devices for extreme scalability and performance. RBD disk images are thinly provisioned, support both read-only snapshots and writable clones, and can be asynchronously mirrored to remote Ceph clusters in other data centers for disaster recovery or backup, making Ceph RBD the leading choice for block storage in public/private cloud and virtualization environments.

File: src/en/discover/technology/index.html [hunk 4]
```diff
@@ -164,15 +162,14 @@ <h2 class="h2 mb-12 lg:mb-16" id="file">File system</h2>
   <div class="grid lg:grid--cols-2 grid--gap-14 lg:grid--gap-28">
     <div>
       <p class="standout">
-        Traditional file system interface (CephFS) with POSIX compliance allowing for detailed metadata, dynamic rebalancing and high
-        performance within the same scalable system as object and block storage.
+      The Ceph File System (CephFS) provides a POSIX-compliant filesystem as a service that is layered on top of the object-based Ceph Storage Cluster. CephFS files are mapped to objects stored by Ceph in the Ceph Storage Cluster. Ceph clients mount CephFS filesystems as kernel objects or as Filesystems in User Space (FUSE).
```
Comment (liewegas) [inline src/en/discover/technology/index.html:165]: The Ceph File System (CephFS) is a robust, fully-featured POSIX-compliant distributed filesystem as a service with snapshots, quotas, and multi-cluster mirroring capabilities. CephFS files are striped across objects stored by Ceph for extreme scale and performance. Linux systems can mount CephFS filesystems natively, via a FUSE-based client, or via an NFSv4 gateway.

File: src/en/discover/technology/index.html [hunk 5]
```diff
@@ -212,23 +209,35 @@ <h3 class="h3">Lorem ipsum</h3>
...
       <p class="p">
-        Through RADOS, organizations can store and access object, block and file types in one unified storage cluster, making data and
-        monitoring easy to manage. Each one of your applications can use the object, block or file system interfaces to the same RADOS
-        cluster simultaneously, which means your Ceph storage system serves as a flexible foundation for all of your data storage needs.
-        Additionally, Ceph’s RADOS provides you with extraordinary data storage scalability, able to store petabytes to exabytes of data,
-        using widely aware hardware.
+	RADOS (Reliable Autonomic Distributed Object Store) seeks to leverage
+	device intelligence to distribute the complexity surrounding consistent
+	data access, redundant storage, failure detection, and failure recovery
+	in clusters consisting of many thousands of storage devices. This is
```
Comment (liewegas) [inline src/en/discover/technology/index.html:227]: s/This/RADOS/

File: src/en/discover/technology/index.html [hunk 6]
```diff
@@ -212,23 +209,35 @@ <h3 class="h3">Lorem ipsum</h3>
...
+	designed for storage systems at the petabyte scale: such systems are
+	necessarily dynamic as they incrementally grow and contract with the
+	deployment of new storage and the decommissioning of old devices. RADOS
+	ensures that there is a consistent view of the data distribution while
+	maintaining consistent read and write access to data objects, and it
+	does this by means of a versioned cluster map.
```
Comment (liewegas) [inline src/en/discover/technology/index.html:233]: drop the trailing ", and it ..."  part?

**Comment by zdover23** [comment]
All your changes have been incorporated. I have made only one addition, where I added a short relative clause to the RBD section in order to make it unambiguous to a first-time reader where the verb in the sentence is. ("Ceph RBD (RADOS Block Device) block storage stripes virtual disks over objects in the Ceph storage cluster, distributing data and workload across all available devices for extreme scalability and performance." is the sentence, and the adjacency of "storage" and "stripes" strikes me as potentially leading a first-time reader into a garden-path sentence.) Anyway, this is already too much exposition of this change.

File: src/en/discover/technology/index.html [hunk 7]
```diff
@@ -42,10 +37,10 @@ <h2 class="color-white h1">Infinitely scalable data storage for object, block an
 
 <section class="section">
   <div class="max-w-192 mx-auto text-center">
-    <h2 class="h2">Store Data in the Cloud</h2>
+    <h2 class="h2">The Ceph Stack -- Architectural Overview</h2>
+    <a href="#object"><img alt="" src="/assets/stack.png" /></a>
```
Comment (adamduncan) [inline src/en/discover/technology/index.html:41]: @zdover23 This link and image was introduced to the page here. What was the desired outcome here? Is this needed in the new site? What is it attempting to link to? Has the asset been added to our `/src/assets/bitmaps` directory?

**Comment by zdover23** [comment]
This is an oversight on my part. I forgot to add the file when I was making
the PR.

I'll make a new PR.

Thanks for jogging my memory.

**Comment by adamduncan** [comment]
@zdover23 No problemo. Grab us if you need anything.

File: src/en/discover/technology/index.html (hunk 7 above)
Comment (zdover23) [reply src/en/discover/technology/index.html:41]: > @zdover23 This link and image was introduced to the page here. What was the desired outcome here? Is this needed in the new site? What is it attempting to link to? Has the asset been added to our `/src/assets/bitmaps` directory?

@adamduncan: I think I've got it fixed here: https://github.com/ceph/ceph.io/pull/177

**Pull request merged**